cmd/demo/opasupport/opa_support_test.go

```diff
@@ -11,22 +12,25 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/mock"
 	"io"
-	"io/ioutil"
 	"log"
 	"net"
 	"net/http"
+	"net/http/httptest"
 	"testing"
 )
 
 type MockClient struct {
 	mock.Mock
-	response []byte
-	err      error
+	response    []byte
+	err         error
+	requestBody []byte
 }
 
-func (m *MockClient) Do(_ *http.Request) (*http.Response, error) {
-	r := ioutil.NopCloser(bytes.NewReader(m.response))
-	return &http.Response{StatusCode: 200, Body: r}, m.err
+func (m *MockClient) Do(r *http.Request) (*http.Response, error) {
+	reqBody, _ := io.ReadAll(r.Body)
+	m.requestBody = reqBody
```

Comment:
Capturing the request body so we can inspect the opa query `method` in the body.